slice/internal/topology/validation.go

```diff
@@ -29,18 +34,48 @@ func AnyAssignment(admission *kueue.Admission) bool {
 
 // AllAssignmentsValid ensures each PodSetAssignment which has a TopologyAssignment
 // defined the TPUSubBlock topology level.
-func AllAssignmentsValid(admission *kueue.Admission) bool {
+func AllAssignmentsValid(admission *kueue.Admission, nodes map[string]corev1.Node) bool {
 	for _, psa := range admission.PodSetAssignments {
 		if psa.TopologyAssignment == nil {
 			continue
 		}
-		if !IsAssignmentValid(psa) {
+		if !IsAssignmentValid(psa, nodes) {
 			return false
 		}
 	}
 	return true
 }
 
-func IsAssignmentValid(psa kueue.PodSetAssignment) bool {
-	return psa.TopologyAssignment != nil && SubblockLevelIndex(psa.TopologyAssignment) != -1
+func IsAssignmentValid(psa kueue.PodSetAssignment, nodes map[string]corev1.Node) bool {
+	if psa.TopologyAssignment == nil {
+		return false
+	}
+
+	if LevelIndex(psa.TopologyAssignment, core.TPUSubBlockLabel) != -1 {
+		return true
+	}
+
+	hostnameLevelIndex := LevelIndex(psa.TopologyAssignment, corev1.LabelHostname)
+	if hostnameLevelIndex == -1 {
+		return false
+	}
+
+	for _, domain := range psa.TopologyAssignment.Domains {
+		if hostnameLevelIndex >= len(domain.Values) {
+			return false
+		}
```

Comment:
when would this happen? would this have to be bug in TAS?

Reply:
It shouldn't happen, but this ensures we prevent a panic here.

Reply:
I would like to keep it just to ensure we never get a panic.

Reply:
Since this is an invariant which shouldn't happen, I'm okay with the program panicking here. If there were some major problem with TAS, it would be better to catch it here than to silently fail

As an alternative we could add a log line, but I'm not against the panic.